tcg/tcg-op-vec.c

```diff
@@ -293,76 +428,98 @@ void tcg_gen_stl_vec(TCGv_vec r, TCGv_ptr b, TCGArg o, TCGType low_type)
 
     tcg_debug_assert(low_type >= TCG_TYPE_V64);
     tcg_debug_assert(low_type <= type);
+
+    uint64_t length;
+
+    switch (low_type) {
+        case TCG_TYPE_V64:
+            length = 8;
+            break;
+        case TCG_TYPE_V128:
+            length = 16;
+            break;
+        case TCG_TYPE_V256:
+            length = 32;
+            break;
+        default:
+            g_assert_not_reached();
+    }
+
+    gen_helper_sym_store_host(tcgv_vec_expr(r), b, tcg_constant_i64(o), tcg_constant_i64(length));
+
     vec_gen_3(INDEX_op_st_vec, low_type, 0, ri, bi, o);
 }
 
 void tcg_gen_and_vec(unsigned vece, TCGv_vec r, TCGv_vec a, TCGv_vec b)
 {
+    vec_vec_op_instrumentation(vece, r, a, b, gen_helper_sym_and_vec);
     vec_gen_op3(INDEX_op_and_vec, 0, r, a, b);
 }
 
 void tcg_gen_or_vec(unsigned vece, TCGv_vec r, TCGv_vec a, TCGv_vec b)
 {
+    vec_vec_op_instrumentation(vece, r, a, b, gen_helper_sym_or_vec);
     vec_gen_op3(INDEX_op_or_vec, 0, r, a, b);
 }
 
 void tcg_gen_xor_vec(unsigned vece, TCGv_vec r, TCGv_vec a, TCGv_vec b)
 {
+    vec_vec_op_instrumentation(vece, r, a, b, gen_helper_sym_xor_vec);
     vec_gen_op3(INDEX_op_xor_vec, 0, r, a, b);
 }
 
 void tcg_gen_andc_vec(unsigned vece, TCGv_vec r, TCGv_vec a, TCGv_vec b)
 {
-    if (TCG_TARGET_HAS_andc_vec) {
+   /* if (TCG_TARGET_HAS_andc_vec) {
         vec_gen_op3(INDEX_op_andc_vec, 0, r, a, b);
```

Comment:
Why is this commented? Is vec_gen_op3 declared above not working (for some operations?) ? 
Same for the 3  others:
INDEX_op_andc_vec
INDEX_op_orc_vec
INDEX_op_nand_vec
INDEX_op_nor_vec
If they need to be implemented ? 
You mention those in section 6.1.7 of the thesis, but I still don't get why the direct support isn't used when the target supports it?